src/arche/rules/result.py

```diff
@@ -40,7 +40,11 @@ class Message:
     summary: str
     detailed: Optional[str] = None
     errors: Optional[Dict[str, Set]] = None
-    _err_keys: Optional[Set[Union[str, int]]] = field(default_factory=set)
+
+    # expression "field(default_factory=set)" has type "Set[_T]", so we have to cast
+    _err_keys: Optional[Set[Union[str, int]]] = cast(
```

Comment:
I am not sure why since in the docs everything is ok https://mypy.readthedocs.io/en/latest/additional_features.html
It maybe be because of Optional.